### PR TITLE
[MOB-475] Fix in-app persistency

### PR DESCRIPTION
**Pull request by vbabenkoru**
Also, trigger in-app queue update on app launch

### CODE REVIEW

File: iterableapi/src/test/java/com/iterable/iterableapi/IterableInAppFileStorageTest.java [hunk 1]
```diff
@@ -39,4 +39,29 @@ public void testInAppPersistence() throws Exception {
         storage = new IterableInAppFileStorage(RuntimeEnvironment.application);
         assertEquals(0, storage.getMessages().size());
     }
+
+    @Test
+    public void testInAppImplicitSave() throws Exception {
+        IterableInAppFileStorage storage = new IterableInAppFileStorage(RuntimeEnvironment.application);
+        IterableInAppMessage testInAppMessage = InAppTestUtils.getTestInAppMessage();
+        storage.addMessage(testInAppMessage);
```
Comment (davidtruong) [inline iterableapi/src/test/java/com/iterable/iterableapi/IterableInAppFileStorageTest.java]: Did you forget to add an assertion for this test case?

Comment (vbabenkoru) [reply iterableapi/src/test/java/com/iterable/iterableapi/IterableInAppFileStorageTest.java]: No, there are assertions below this line. The first three lines make it persist a message, and the code that follows checks that the message and its properties were stored and retrieved correctly across instances of `IterableInAppFileStorage`.

Comment (vbabenkoru) [reply iterableapi/src/test/java/com/iterable/iterableapi/IterableInAppFileStorageTest.java]: Added comments to the test.

Comment (davidtruong) [reply iterableapi/src/test/java/com/iterable/iterableapi/IterableInAppFileStorageTest.java]: What's the point of creating a new IterableInAppFileStorage(RuntimeEnvironment.application); ?

Comment (vbabenkoru) [reply iterableapi/src/test/java/com/iterable/iterableapi/IterableInAppFileStorageTest.java]: It makes it load the previously persisted data into a new instance of `IterableInAppFileStorage` to check if it's been correctly saved and restored from disk.

**Pull request merged**